ambry-api/src/test/java/com.github.ambry/account/InMemAccountServiceFactory.java

```diff
@@ -0,0 +1,198 @@
+      return accountUpdateConsumers.add(accountUpdateConsumer);
+    }
+
+    @Override
+    public synchronized boolean removeAccountUpdateConsumer(Consumer<Collection<Account>> accountUpdateConsumer) {
+      Objects.requireNonNull(accountUpdateConsumer, "accountUpdateConsumer to unsubscribe cannot be null");
+      return accountUpdateConsumers.remove(accountUpdateConsumer);
+    }
+
+    @Override
+    public synchronized Collection<Account> getAllAccounts() {
+      return idToAccountMap.values();
+    }
+
+    @Override
+    public void close() throws IOException {
+      // no op
+    }
+
+    /**
+     * Clears all the accounts in this {@code AccountService}.
+     */
+    public synchronized void clear() {
+      Collection<Account> allAccounts = idToAccountMap.values();
+      idToAccountMap.clear();
+      nameToAccountMap.clear();
+      for (Consumer<Collection<Account>> accountUpdateConsumer : accountUpdateConsumers) {
+        accountUpdateConsumer.accept(allAccounts);
+      }
+    }
+
+    /**
+     * Creates and adds an {@link Account} to this {@link AccountService}. The account will contain one container
+     * with {@link Container#DEFAULT_PUBLIC_CONTAINER_ID}, one with {@link Container#DEFAULT_PRIVATE_CONTAINER_ID} and
+     * one other random {@link Container}.
+     * @return the {@link Account} that was created and added.
+     */
+    public synchronized Account createAndAddRandomAccount() {
+      short refAccountId;
+      String refAccountName;
+      do {
+        refAccountId = Utils.getRandomShort(TestUtils.RANDOM);
+        refAccountName = UtilsTest.getRandomString(10);
+      } while (idToAccountMap.containsKey(refAccountId) || nameToAccountMap.containsKey(refAccountName));
+      Account.AccountStatus refAccountStatus = Account.AccountStatus.ACTIVE;
+      Container randomContainer = getRandomContainer(refAccountId);
+      Container publicContainer = getDefaultContainer(refAccountId, false);
+      Container privateContainer = getDefaultContainer(refAccountId, true);
+      Account account = new AccountBuilder(refAccountId, refAccountName, refAccountStatus,
+          Arrays.asList(publicContainer, privateContainer, randomContainer)).build();
+      updateAccounts(Collections.singletonList(account));
+      return account;
+    }
+
+    /**
+     * Creates and returns default public/private container for {@code accountId}.
+     * @param accountId the account id for the container
+     * @param isPrivate {@code true} if private container is required. {@code false} for public.
+     * @return default public/private container for {@code accountId}.
+     */
+    private Container getDefaultContainer(short accountId, boolean isPrivate) {
+      Container container;
+      if (isPrivate) {
+        container =
+            new ContainerBuilder(Container.DEFAULT_PRIVATE_CONTAINER_ID, Container.DEFAULT_PRIVATE_CONTAINER_NAME,
```

Comment:
The copy builder constructor might make this easier: 
`new ContainerBuilder(Container.DEFAULT_PRIVATE_CONTAINER).setParentAccountId(accountId).build()`

Since the only thing different from the default container is the account id.

Reply:
Good suggestion - eliminated the need to have a different function for it